tools/biddersettings.html

```diff
@@ -0,0 +1,248 @@
+        }
+        label {
+            margin: 5px 0px;
+            display: block;
+            padding: 0px;
+            font-size: 18px;
+        }
+        div#successBanner {
+            background: #e0ffdc;
+            font-size: 18px;
+            font-weight: bolder;
+            padding: 8px;
+            margin: 14px 0px;
+            border-radius: 8px;
+            border: 1px solid #009900;
+            color: #009900;
+        }
+        .errorTxt {
+            color: #CC0000;
+            font-weight: bolder;
+            background: #FFFEC8;
+        }
+        .successTxt {
+            background: #e0ffdc;
+        }
+        .showSuccess {
+            display: inline-block;
+        }
+        .hideSuccess {
+            display: none;
+        }
+        -->
+    </style>
+    <script type="text/javascript">
+        function convertBidderSettings () {
+
+            removeOutputStyles();
+
+            var inputObj = document.getElementById('settings_txt');
+            var bidSetStr = inputObj.value;
+
+            if (!bidSetStr) {
+                setOutputText("Please paste your original bidderSettings Javascript code above, then click Convert...", true);
+                return;
+            }
+
+            bidSetStr = prepString(bidSetStr);
+
+            try {
+                var bidSetObj = eval( "(" + bidSetStr + ")" );
+            } catch (e) {
+                console.log("OOOPS! JS CONVERSION ERROR: ");
+                console.log(e);
+                handleError(e.message);
+                return;
+            }
+
+            inputObj.value = "var bidderSettings = " + bidSetStr;
+
+            bidSetObj = {
+                bidderSpec: { bids: ["// Placeholder - Replace this with your bids array"] },
+                bidderSettings: bidSetObj
+            };
+
+            var jsonStr = JSON.stringify(bidSetObj, function (key, value) {
+                if (typeof value === 'function' || value instanceof Function) {
+                    var arr = getFunctionArray(value);
+                    var arr2 = ["valueIsFunction"].concat(arr);
+                    return arr2;
+                }
+                return value;
+            }, 2);
+
+            setOutputText(jsonStr);
+        }
+
+        function prepString (str) {
+            str = str.trim();
+            var i = str.indexOf("{");
+            if (i > 0) {
+                str = str.substring(i);
+            }
+            i = str.lastIndexOf("}");
+            str = str.substring(0, i + 1);
+            str = str.replace(/“+|”+/g, '"');                       // Globally replace smart quotes with normal "
+            return str;
+        }
+
+        function handleError (msg) {
+            var str = "DOH! Something is wrong with your code.";
+
+            // Ignore this error - it's from the parens we use to wrap the string before doing eval() to convert the input string to JS
+            if (msg != "Unexpected token )") {
```

Comment:
Yes - this is a particular error message that happens in a case which should not be reported to the user because it's internal in nature. Good question, though!